### DISCUSSION TITLE
Upload code to XIAO ESP32-C3

### DISCUSSION THREAD

**Discussion by Mishko959** [thonny/thonny#3143] [General]
Hi, I'm a total beginner and I have a problem I hope you can help me with.

I have been using XIAO NRF52480 board with CircuitPython code which works well for my needs, but recently I have bought XIAO ESC32-C3 boards and would like to upload the same code , but I found out I can't just copy the files to the CIRCUIRPY folder as I've been doing with other boards. After doing some research, I've stumbled upon some options, like Adafruit Esptool, and Thonny was also mentioned as one of he possibilities. I have a number of ESP boards I have to program and send along as a part of a bigger project as soon as possible, but I'm stuck with this problem I haven't expected, and I have no time to change the boards for some other type. 

Can you please let me know if I can indeed use Thonny to upload the code, and if so, explain how exactly? Thank you very much in advance for any advice.

Miso

**Comment by aivarannamaa** [comment]
It looks like CircuitPython is not ready for ESP32-C3 boards: https://github.com/adafruit/circuitpython/issues/8658

**Reply by Mishko959** [reply]
Thanks for the reply. Yet, they offer it for download here https://circuitpython.org/board/seeed_xiao_esp32c3/. I've been playing with this for almost two weeks now, and this causes me a really big problem. As I said above, I previously used XIAO NRF52840, and this Circuit Python program, which in essence only reads the pressure from BMP280 sensor via I2C and drives two LEDs worked without problems, although that was my first attempt at Circuit Python. And now I'm sitting on 150 XIAO ESP32C3 boards and don't know what to do.

There are only 3 files, the code itself and two files from Adafruit in a separate lib folder. I even managed to upload some of them from Thonny, but never all three, because either the connection is lost, or something else happens. So now I'm trying to make and upload an Arduino sketch, but that also doesn't seem to work as it should, and I'm stuck in a cycle of uploading sketches, getting error messages, reuploading original firmware via EspRFTeestTool and so on. And I just have no time to change the board, and I'm already so behind that the customer will just cancel the order and ask for refund.

So, you think it makes no sense to insist on Circuit Python with this board?

Thanks,

Miso

**Reply by aivarannamaa** [reply]
I don't think "insisting" is a good idea in open source. 

Have you tried MicroPython instead of CircuitPython? There are couple of C3 builds, which may be worth trying: https://micropython.org/download/?mcu=esp32c3

**Reply by Mishko959** [reply]
Thanks for the hint. Will try and report back. "Insisting" in this context was meant as further loss of time for dubious return:)
